parsers/GB_NIR.py

```diff
@@ -0,0 +1,245 @@
+                )  # remaining generation
+
+                production_list.append(
+                    {
+                        "zoneKey": zone_key,
+                        "datetime": _parse_effective_time(
+                            generation_row["EffectiveTime"]
+                        ),
+                        "production": {
+                            "biomass": None,
+                            "coal": None,
+                            "gas": None,
+                            "hydro": None,
+                            "nuclear": None,
+                            "oil": None,
+                            "solar": None,
+                            "wind": wind_generation_mw,
+                            "geothermal": None,
+                            "unknown": unknown_generation_mw,
+                        },
+                        "source": "smartgriddashboard.com",
+                    }
+                )
+            else:
+                return production_list
+
+        return production_list
+
+    except TypeError as e:
+        ParserException(
+            "GB_NIR.py", f"Failed to retrieve production at requested timestamp: {e}"
+        )
+        raise TypeError
+
+
+@refetch_frequency(timedelta(minutes=15))
+def fetch_exchange(
+    zone_key1: str = "GB",
+    zone_key2: str = "GB-NIR",
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> dict:
+    sortedZoneKeys = "->".join(sorted([zone_key1, zone_key2]))
+    moyle_json = _fetch_json_data("NI", "interconnection", session, target_datetime)
+
+    interconnection_list = []
+
+    if sortedZoneKeys == "GB->GB-NIR":
+        for moyle_row in moyle_json:
+            if moyle_row["Value"]:
+                interconnection_list.append(
+                    {
+                        "sortedZoneKeys": sortedZoneKeys,
+                        "datetime": _parse_effective_time(moyle_row["EffectiveTime"]),
+                        "netFlow": moyle_row["Value"],
+                        "source": "smartgriddashboard.com",
+                    }
+                )
+            else:
+                return interconnection_list
+
+    elif sortedZoneKeys == "GB-NIR->IE":
+        production_json = _fetch_json_data(
+            "NI", "generationactual", session, target_datetime
+        )
+
+        demand_json = _fetch_json_data("NI", "demandactual", session, target_datetime)
+
+        for production_row, demand_row, moyle_row in zip(
+            production_json, demand_json, moyle_json
+        ):
+            production_mw = production_row["Value"]
+            demand_mw = demand_row["Value"]
+            moyle_mw = moyle_row["Value"]
+
+            if production_mw and demand_mw and moyle_mw:
+                power_flow_mw = (
+                    production_mw + moyle_mw - demand_mw
+                )  # power flow gb_nir -> ie
```

Comment:
I'm not sure we can do this as we don't account for losses here and would get a higher exchange than it should be.
@unitrium could I get your opinion on this?

Reply:
To my knowledge the SONI / EirGrid dashboard should be taking into account the demand side system losses, but is there something else that should be taken into account to compute these flows? The dashboard has also the nubmers for ROI available (demand, generation and EWIC transfers), so it might be interesting to compare the two and see if there's a more robust way to compute the power flows between them.